python/fio/fio_plugin.py

```diff
@@ -0,0 +1,65 @@
+import sys
+import typing
+import json
+import subprocess
+import dataclasses
+from traceback import format_exc
+from typing import Union
+import configparser
+from pathlib import Path
+
+from arcaflow_plugin_sdk import plugin
+from fio_schema import FioParams, FioSuccessOutput, FioErrorOutput, fio_output_schema
+
+
+def build_input_config(params: FioParams, filepath: Path):
+    cfg = configparser.ConfigParser()
+    cfg[params.name] = {}
+    for key,value in dataclasses.asdict(params).items():
+        if key != 'name':
+            cfg[params.name][key] = str(value)
+    with open(filepath, 'w') as temp:
+        cfg.write(temp, space_around_delimiters=False)
+
+
+@plugin.step(
+    id="workload",
+    name="fio workload",
+    description="run an fio workload",
+    outputs={"success": FioSuccessOutput, "error": FioErrorOutput}
+)
+def run(params: FioParams) -> typing.Tuple[str, Union[FioSuccessOutput, FioErrorOutput]]:
+    outfile_temp_path = Path('fio-plus.json')
+    infile_temp_path = Path('fio-input-tmp.job')
+
+    try:
+        build_input_config(params, infile_temp_path)
+        cmd = [
+            'fio',
+            f'{infile_temp_path}',
+            '--output-format=json+',
+            f"--output={outfile_temp_path}"
+        ]
+
+        try:
+            subprocess.check_output(cmd)
+        except subprocess.CalledProcessError:
+            return 'error', FioErrorOutput(format_exc())
+
+        infile_temp_path.unlink(missing_ok=True)
```

Comment:
I prefer having a cleanup function that can be called by default but also can leave the temporary files in place for debugging purposes when cleanup=false e.g. 
Just my 2 cents.